bin/fin

```diff
@@ -3636,23 +3636,19 @@ mysql_db_create ()
 	local __dump_password="${dbpassword:-\$MYSQL_ROOT_PASSWORD}"
 	local __db="${ARGV[0]}"
 	local __db_user="\$MYSQL_USER"
-	local __db_charset="${dbuser:-utf8}"
-	local __db_collation="${dbuser:-utf8_general_ci}"
+	local __db_charset="${dbcharset:-utf8}"
+	local __db_collation="${dbcollation:-utf8_general_ci}"
 
-	[[ "$__db" == "" ]] && echo-error "Provide a database name to create." && exit 1
+	[[ "${__db}" == "" ]] && echo-error "Provide the database name to create" && exit 1
 
 	# set db to space to avoid database parameter for mysql command because it might be nonexistent
-	mysql --db-user="$__dump_user" --db-password="$__dump_password" --db=" " \
-		'CREATE DATABASE `'${__db}'` CHARACTER SET '${__db_charset}' COLLATE '${__db_collation}''
+	_mysql --db-user="${__dump_user}" --db-password="${__dump_password}" \
+		"CREATE DATABASE ${__db} CHARACTER SET ${__db_charset} COLLATE ${__db_collation};" \
+		"GRANT ALL PRIVILEGES ON ${__db}.* TO ${__db_user};"
```

Comment:
I do not really understand the change. I deliberately made GRANT a separate command to distinguish creation issue and grant issue.